core/src/main/java/com/oath/oak/MemoryManager.java

```diff
@@ -6,147 +6,45 @@
 
 package com.oath.oak;
 
-import javafx.util.Pair;
+
 
 import java.nio.ByteBuffer;
-import java.util.ArrayList;
-import java.util.LinkedList;
-import java.util.concurrent.atomic.AtomicLong;
+
 
 public class MemoryManager {
+    private final OakMemoryAllocator keysMemoryAllocator;
     private final OakMemoryAllocator memoryAllocator;
-    private final AtomicLong[] timeStamps;
-    private final ArrayList<LinkedList<Pair<Long, ByteBuffer>>> releasedArray;
-    private final AtomicLong max;
-
-    private static final int RELEASES_DEFAULT = 100; // TODO: make it configurable
-    private final ThreadIndexCalculator threadIndexCalculator;
-
-    private int releases; // to be able to change it for testing
 
-    public MemoryManager(OakMemoryAllocator ma, ThreadIndexCalculator threadIndexCalculator) {
+    public MemoryManager(OakMemoryAllocator ma) {
         assert ma != null;
 
         this.memoryAllocator = ma;
-        this.timeStamps = new AtomicLong[ThreadIndexCalculator.MAX_THREADS];
-        for (int i = 0; i < ThreadIndexCalculator.MAX_THREADS; i++) {
-            this.timeStamps[i] = new AtomicLong(0);
-        }
-        this.releasedArray = new ArrayList<>(ThreadIndexCalculator.MAX_THREADS);
-        for (int i = 0; i < ThreadIndexCalculator.MAX_THREADS; i++) {
-            releasedArray.add(i, new LinkedList<>());
-        }
-        max = new AtomicLong(0);
-        releases = RELEASES_DEFAULT;
-        this.threadIndexCalculator = threadIndexCalculator;
+        keysMemoryAllocator = new DirectMemoryAllocator();
```

Comment:
I think concerns are separated better if you inject `keysMemoryAllocator` as a constructor argument (and allocate a new `DirectMemoryAllocator` if null)